src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManager.cs

```diff
@@ -129,8 +129,14 @@ public override async Task<T> InvokeConnectionAsync<T>(string connectionId, stri
 
             var invocationId = _clientInvocationManager.Caller.GenerateInvocationId(connectionId);
             var message = AppendMessageTracingId(new ClientInvocationMessage(invocationId, connectionId, _callerId, SerializeAllProtocols(methodName, args, invocationId)));
-            await WriteAsync(message);
+            // The ack number of invocation will be set inside `WriteAsync`. So adding invocation should be first.
             var task = _clientInvocationManager.Caller.AddInvocation<T>(_hub, connectionId, invocationId, cancellationToken);
+            await WriteAsync(message);
+            if (ServiceConnectionContainer is not MultiEndpointServiceConnectionContainer)
```

Comment:
sounds tricky, a ServiceLifttimeManager should not have knowledge of what the container type is, it looks like a strong assumption and is fragile. What if we later change to use other container?